### PR TITLE
refactor(plugin-server): Move `vm.*` tests over to ClickHouseLand

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Another step towards ripping Postgres ingestion out, while increasing coverage of the Kafka & ClickHouse pipeline. Modernizes `vm.test.ts`, `vm.timeout.test.ts`, `vm.lazy.test.ts`, and `vm.extra-lazy.test.ts`.

### CODE REVIEW

File: plugin-server/tests/shared/vm.test.ts [hunk 1]
```diff
@@ -46,8 +46,8 @@ describe('vm tests', () => {
     let closeHub: () => Promise<void>
 
     beforeEach(async () => {
-        ;(Client as any).mockClear()
-        ;[hub, closeHub] = await createHub({ KAFKA_ENABLED: false })
+        jest.mocked(KafkaProducerWrapper).mockClear()
```
Comment (macobo) [inline plugin-server/tests/shared/vm.test.ts:49]: Note: I believe this is a dead line of code - mocks are automatically cleared.

File: plugin-server/tests/helpers/clickhouse.ts [hunk 2]
```diff
@@ -30,23 +31,23 @@ export async function resetTestDatabaseClickhouse(extraServerConfig: Partial<Plu
 }
 
 export async function delayUntilEventIngested(
-    fetchEvents: () => Promise<any[] | any>,
+    fetchEvents: () => Promise<Record<string, any>[] | number>,
```
Comment (macobo) [inline plugin-server/tests/helpers/clickhouse.ts:34]: I don't think this type is correct - e.g. `fetchSessionRecordingEvents` returns an array not a object/record and it's being used widely.

File: plugin-server/src/utils/db/kafka-producer-wrapper.ts [hunk 3]
```diff
@@ -70,6 +70,10 @@ export class KafkaProducerWrapper {
         }
     }
 
+    async rawSendMessage(kafkaMessage: ProducerRecord): Promise<void> {
```
Comment (macobo) [inline plugin-server/src/utils/db/kafka-producer-wrapper.ts:73]: Why do we need this util?

File: plugin-server/src/utils/db/hub.ts [hunk 4]
```diff
@@ -292,11 +292,8 @@ export async function createHub(
         hub.mmdbUpdateJob?.cancel()
         await hub.db.postgresLogsWrapper.flushLogs()
         await hub.jobQueueManager?.disconnectProducer()
-        if (kafkaProducer) {
-            clearInterval(kafkaProducer.flushInterval)
-            await kafkaProducer.flush()
-            await kafkaProducer.producer.disconnect()
-        }
+        await kafkaProducer.flush()
```
Comment (macobo) [inline plugin-server/src/utils/db/hub.ts:295]: Move flush within disconnect?

File: plugin-server/tests/helpers/clickhouse.ts (hunk 2 above)
Comment (macobo) [reply plugin-server/tests/helpers/clickhouse.ts:34]: I see I misread the type. Using `Record<string, any>` as a 'this is a row from the database' is confusing imo. Why not any[] or use a generic type argument?

File: plugin-server/src/utils/db/hub.ts [hunk 5]
```diff
@@ -292,11 +292,7 @@ export async function createHub(
         hub.mmdbUpdateJob?.cancel()
         await hub.db.postgresLogsWrapper.flushLogs()
         await hub.jobQueueManager?.disconnectProducer()
-        if (kafkaProducer) {
-            clearInterval(kafkaProducer.flushInterval)
-            await kafkaProducer.flush()
-            await kafkaProducer.producer.disconnect()
-        }
+        await kafkaProducer.disconnect()
```
Comment (macobo) [inline plugin-server/src/utils/db/hub.ts:295]: This is nice!

**Review by macobo — APPROVED** [review]

**Review by Twixes — COMMENTED** [review]
Should be addressed!

**Pull request merged**